pkg/zrouter/zmiddlewares/cache.go

```diff
@@ -20,7 +20,7 @@ func CacheMiddleware(cache zcache.ZCache, config domain.CacheConfig) func(next h
 			path := r.URL.Path
 			fullURL := constructFullURL(r)
 
-			if ttl, found := config.Paths[path]; found {
+			if ttl, found := matchPathWithConfig(path, config.Paths); found {
```

Comment:
What if we do the same here as we do on LoggingMiddleware? We could process paths to regex once outside the fn, and use it on the resulting middleware we actually run per request. We could save some operations then